### ISSUE TITLE
Client documentation should use explicit buildProtocol()s rather than magic "protocol = MyProtocol" on factories

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5044]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5044|
|Type|enhancement|
|Created|2011-04-09 02:03:56Z|
|Branch|https://github.com/twisted/twisted/tree/explicit-buildProtocol-5044|

Observation at the Twisted Lab in Boston suggests that buildProtocol as an important protocol/factory integration point is hard for new users to understand. The client howto should use buildProtocol by default, but also explain the "protocol = ProtocolClass" shortcut.

<details><summary>Searchable metadata</summary>

```
trac-id__5044 5044
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_explicit_buildProtocol_5044 branches/explicit-buildProtocol-5044
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1302314636000000 1302314636000000
changetime__1317777139000000 1317777139000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [32546]) Branching to 'explicit-buildProtocol-5044'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

Ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

The news fragment should be a .doc or a .misc.  Also, is this really a change about "consistency"?  Or explicitness (as the ticket summary and branch name seem to express)?

The code and doc changes themselves look great.  Please merge once the news file reflects something an application developer would want to read about. :)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [32757]) Merge explicit-buildProtocol-5044.
Fixes: [#5044](https://github.com/twisted/twisted/issues/5044)
Author: itamar
Review: exarkun

The clients howto now uses buildProtocol more explicitly, hopefully making it clearer where protocols come from.
